alacritty_terminal/src/tty/windows/mod.rs

```diff
@@ -306,3 +323,28 @@ impl EventedPty for Pty {
         }
     }
 }
+
+impl Drop for Pty {
+    fn drop(&mut self) {
+        if let PtyHandle::Conpty(c) = &self.handle {
+            let conout = &mut self.conout;
+            let done = AtomicBool::new(false);
+
+            // The conpty close call will block until the conout pipe is drained.
+            // So we must spawn a thread which continuously attempts to read the pipe
+            // until the call has finished.
+            crossbeam::thread::scope(|s| {
+                s.spawn(|_| {
+                    let mut buf = [0u8; 1024];
+                    while !done.load(Ordering::SeqCst) {
+                        conout.read(&mut buf).unwrap();
+                    }
+                });
+
+                c.close();
+                done.store(true, Ordering::SeqCst);
+            })
+            .unwrap();
```

Comment:
Wait, why do we need crossbeam and a separate thread now? I thought you said that reordering the drop operations is sufficient since the pipes handle things internally.

We didn't need this previously, why do we need this now?

Reply:
We *can* live without spawning this thread. This is because, yes, the `EventedAnonRead` pipe is draining the pseudoconsole pipe into its own internal buffer.

But that's not guaranteed to always work:
- If we ever change the pipe implementation (e.g. implement #1905), then this might break.
- It's possible there will be so much to drain that the `EventedAnonRead` pipe won't be able to fit it all in it's internal buffer.

Using a scoped thread and explicitly draining the whole thing is intended to make the code robust against those situations. (I used crossbeam so that the scoped thread can reference `conout`.)

If you prefer I can remove this thread and leave a comment in the code documenting `EventedAnonRead` currently handles the draining issue.

Reply:
(This possible need for extra defensiveness only struck me this morning when preparing this PR)

Reply:
> It's possible there will be so much to drain that the EventedAnonRead pipe won't be able to fit it all in its internal buffer.

How? I thought that there's always only one event sent after closing it.

Pulling in crossbeam just for this seems absolutely crazy to me. Especially because I do not think this kind of complexity is a good fit for a destructor. I'd assume that this might actually make the code less robust, because it so, so much harder to reason about things.

Reply:
> How? I thought that there's always only one event sent after closing it.

All [the docs for `ClosePseudoConsole`](https://docs.microsoft.com/en-us/windows/console/closepseudoconsole) say is "a final painted frame". There's no guarantee how much data that is.

> Pulling in crossbeam just for this seems absolutely crazy to me. Especially because I do not think this kind of complexity is a good fit for a destructor. I'd assume that this might actually make the code less robust, because it so, so much harder to reason about things.

Ok. Let's just scrap this bit and go back to just fixing the drop order.